### PR TITLE
NMS-14470: Don't persist resources that are marked as such

### PR DESCRIPTION
**Pull request by amay94**
### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-14470

### CODE REVIEW

**Review by cgorantla — COMMENTED** [review]
Nice!  looks good to me.

Possible to add specific test case for skipping collection sets ?

**Comment by amay94** [comment]
It seems straightforward, but I'm struggling to work out a way to mock this up for a unit test. The `CollectionSetBuilder` doesn't take anything implementing `Persister` as an argument, and mocking the `CollectionSetDTO` (where `Persister` classes are actually introduced) looks like it would require some changes to the visibility of fields within that class.

**Comment by cgorantla** [comment]
> It seems straightforward, but I'm struggling to work out a way to mock this up for a unit test. The `CollectionSetBuilder` doesn't take anything implementing `Persister` as an argument, and mocking the `CollectionSetDTO` (where `Persister` classes are actually introduced) looks like it would require some changes to the visibility of fields within that class.

May be achievable through `KafkaPersisterIT`  but it looks like this should work!

**Review by cgorantla — APPROVED** [review]

**Comment by amay94** [comment]
I managed to get a unit test working. I originally had the `CollectionSetMapper` returning null if the resulting `CollectionSetProto` had no resources, but I changed that so it just returns an empty collection set. Now, the `KafkaPersister` decides to send/not send based on the number of resources within the `CollectionSet`.

**Review by thomasbigger584 — COMMENTED** [review]
Looks good to me based on the jira requirements, although i'm not too familiar with this area of the codebase yet

File: features/kafka/producer/src/main/java/org/opennms/features/kafka/producer/collection/KafkaPersister.java [hunk 1]
```diff
@@ -54,20 +56,37 @@ public class KafkaPersister implements Persister {
...
     private String topicName = "metrics";
 
+    public KafkaPersister(ServiceParameters params) {
+        m_params = params;
+    }
+
+    /**
+     * Only for Mockito, DO NOT USE OTHERWISE.
+     */
+    public KafkaPersister() {
```
Comment (cgorantla) [inline features/kafka/producer/src/main/java/org/opennms/features/kafka/producer/collection/KafkaPersister.java:72]: Is this getting used somewhere ?  Why can't it be handled with other constructor ?

**Review by aramosvizcarra — DISMISSED** [review]
LGTM!

**Review by cgorantla — APPROVED** [review]

**Pull request merged**